docs/getting-started/0-agent.md

```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
+  try {
+    const response = await axios.get(genesisUrl); //Axios is a lightweight HTTP client, you can use any other library
+    return response.data;
+  } catch (e) {
+    console.log('EXCEPTION> downloadGenesis:', e);
+    return null;
+  }
+}
+
+
+//Function for storing genesis string value to a local file on the device 
+async function storeGenesis(
+  genesis: string,
+  fileName: string,
+): Promise<string> {
+  try {
+    const genesisPath = `${RNFS.DocumentDirectoryPath}/${fileName}`;
+    await RNFS.writeFile(genesisPath, genesis, 'utf8');
+    return genesisPath;
+  } catch (e) {
+    console.log('EXCEPTION> storeGenesis:', e);
+    return '';
+  }
+}
+
+//Somewhere in the code (before init Agent)
+const genesis = await downloadGenesis(GENESIS_URL);
+const genesisPath = await storeGenesis(genesisString, 'genesis.txn');
+
+```
+
+Note: You do not need the genesis file if you are creating a connection between your Agent and another Agent for exchanging simple messages.
+
+### 3- Get Mediator Connection URL (Optional)
+
+Mediators (Routing Agents) are Agents that serve as intermediaries to facilitate the flow of messages between other types of agents.
+You will need a mediator Agent if you are going to deal with VC (Verifiable Credentials), however, you can ignore the mediator step if you are creating an Agent for the sole purpose of exchanging messages with another Agent. 
+
+Example: If you are testing VC related functionality and need a mediator, you can use the [Indicio Public Mediator](https://indicio-tech.github.io/mediator/).
+- Head to [Indicio Public Mediator](https://indicio-tech.github.io/mediator/).
+- Copy mediator invite url and save it (i.e. MEDIATOR_INVITE = "url").
+
+Note: If the invite url uses insecure http you will have to adjust your code to allow for insecure traffic.
+- Instructions for [iOS](https://stackoverflow.com/questions/31254725/transport-security-has-blocked-a-cleartext-http).
+- Instructions for [Android](https://stackoverflow.com/questions/51902629/how-to-allow-all-network-connection-types-http-and-https-in-android-9-pie).
+
+More about [Mediators](3-routing.md).
+
+### 4- Create Agent
+
+```ts
+const agentConfig: InitConfig = {
+      label: 'my-agent9',
+      mediatorConnectionsInvite: MEDIATOR_INVITE,
+      walletConfig: {id: 'walletId10'},
+      walletCredentials: {key: 'testkey0230482304823042304244'},
+      autoAcceptConnections: true,
+      poolName: 'test-193',
+      genesisPath,
+      logger: new ConsoleLogger(LogLevel.debug),
+    };
+
+    const agent = new Agent(agentConfig, agentDependencies);
+
+```
+
+### 5- Create Transports
+
+After creating the Agent object, we need to create and set the Inbound/Outbound transports that will handle traffic to/from the Agent.
+
+
+```ts
+  const httpOutboundTransporter = new HttpOutboundTransporter();
+  agent.setOutboundTransporter(httpOutboundTransporter);
+
+  //Inbound transports are currently built-in, you don't need to create them.
+```
```

Comment:
```suggestion

### Outbound transports

Outbound transport are for sending messages. Outbound transports can optionally be used for receiving messages (e.g. in the case of a mediator). Currently two transports are supported as part of the framework, HTTP and WebSockets.

\`\`\`ts
  // For sending messages to http endpoints
  const httpOutboundTransport = new HttpOutboundTransport();
  agent.registerOutboundTransport(httpOutboundTransport);
  
  // For sending messages to ws endpoints
  const wsOutboundTransport = new WsOutboundTransport();
  agent.registerOutboundTransport(wsOutboundTransport);
\`\`\`

### Inbound transports

Inbound transports are for receiving messages. The inbound transports are only available in NodeJS environments, as we need to start a server to expose an HTTP endpoint and WebSocket server. You can import the inbound transports from the `@aries-framework/node` package

\`\`\`ts
  // For receiving messages on an HTTP endpoint
  const httpInboundTransport = new HttpInboundTransport({ port: 3000 });
  agent.registerInboundTransport(httpInboundTransport);
  
  // For receiving messages on an WS endpoint
  const wsInboundTransport = new WsInboundTransport({ port: 30001 });
  agent.registerInboundTransport(wsInboundTransport);
\`\`\`
```

Reply:
Please remove the backticks, but github was struggling with the backtikcs in the suggestions